packages/expect/src/asymmetricMatchers.ts

```diff
@@ -109,6 +107,7 @@ class Anything extends AsymmetricMatcher {
 
 class ArrayContaining extends AsymmetricMatcher {
```

Comment:
Looking at these, I wonder if `AsymmetricMatcher` should take a generic that is the type of `sample`?

Reply:
Hmm, I'm not sure how this would be implemented. Do you have an example?

Reply:
See https://github.com/facebook/jest/pull/7919/commits/bf24385dd5079c58656a45b05b29c91bae85eab9